### PR TITLE
Fix #57

### LINKED ISSUE #57: TestFailure: TestListeners.testBasic
```
testBasic(org.antlr.v4.test.runtime.go.TestListeners)  Time elapsed: 2.195 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<(a 1 2)
[1
2
]> but was:<(a 1 2)
[]>
    at org.junit.Assert.assertEquals(Assert.java:115)
    at org.junit.Assert.assertEquals(Assert.java:144)
    at org.antlr.v4.test.runtime.go.TestListeners.testBasic(TestListeners.java:54)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:622)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
    at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
    at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
```

### PR DESCRIPTION
**Pull request by pboyer**
Fix for #57 

@willfaught PTAL

### CODE REVIEW

File: runtime/Go/antlr/tree.go [hunk 1]
```diff
@@ -45,6 +45,8 @@ type TerminalNode interface {
 
 type ErrorNode interface {
 	TerminalNode
+
+	errorNode()
```
Comment (willfaught) [inline runtime/Go/antlr/tree.go]: What's the motivation for this?

Comment (willfaught) [inline runtime/Go/antlr/tree.go:49]: Can you please explain how this fixes #57?

Comment (pboyer) [reply runtime/Go/antlr/tree.go:49]: Before this change, an `ErrorNode` and `TerminalNode` were indistinguishable.

**Comment by willfaught** [comment]
How was that causing the test failure?

**Comment by pboyer** [comment]
@willfaught This test created a custom `ParseTreeListener`. This data structure implemented a `VisitTerminal` method that printed the node. The `VisitErrorNode` method was empty. 

Before this change, an `ErrorNode` and `TerminalNode` were type-wise identical. Note the implementation of the `ErrorNode` interface. So, the `VisitErrorNode` method was called (see the type switch) and the node was not printed. In Java, these would be distinguishable. Go's duck typing like semantics cause this issue.

A common approach to this problem is to have a method that distinguishes the interface. See for example: https://github.com/golang/go/blob/master/src/go/ast/ast.go#L491

This is the approach I took here.

**Comment by willfaught** [comment]
I was interested more in where in the code the type confusion was causing the problem. In any case the change makes sense and LGTM!

**Pull request merged**